core/psh/sysexec.c

```diff
@@ -25,15 +27,91 @@ void psh_sysexecinfo(void)
 }
 
 
+static int psh_sysexec_wldargmatch(const char *wld, int argc, char **argv, char const *wldend)
```

Comment:
maybe we could sort arguments - `const char *wld, char const *wldend, int argc, char **argv`?

I would still prefer to have `const char* wld, size_t wldlen` as this is standard API in C library for "passing buffer which might not contain `\0`"

Reply:
Also: please rename function and variable names, this has nothing to do with wildcards

Reply:
The `wldlen` approach is implemented. Function names redefined together with macro and filename